app/src/main/java/com/example/habithelper/activities/HabitDetailsActivity.java

```diff
@@ -95,38 +108,59 @@ public void done(List<TrackDay> daysTracked, ParseException e) {
                     return;
                 }
                 numDaysTracked = daysTracked.size();
-
                 // ensures the number of days the user has tracked is greater than the number of habits they are tracking
                 // required for LeastSquaresRegression
                 if (numDaysTracked > numHabits) {
                     averageMood = getAverageMood(daysTracked);
-                    //TODO: set visibility of percentage calculator!
                     lrc = new LinearRegressionCalculator();
                     // calculates the array of correlation coefficients of each habit on the user's mood
                     lrc.performLeastSquares(daysTracked, numDaysTracked, numHabits);
                     double[] resultsArray = lrc.leastSquaresResult;
                     int habitIndex = findIndexOfHabit();
-                    for (int i = 0; i< resultsArray.length; i++){
-                        System.out.println(resultsArray[i]);
+                    if (habitIndex != -1) {
+                        displayPercentChange(resultsArray, habitIndex);
+                    } else {
+                        // error has occurred
+                        clPercent.setVisibility(View.GONE);
```

Comment:
These 2 lines are the same as the else statement below. Maybe encapsulate them into a method like showNoDataView() or showNoDataError() (or similar)